src/ui/src/builder/BuilderJournal.vue

```diff
@@ -1,11 +1,205 @@
+	timestamp: string;
+	blockOutputs: Record<string, BlockOutput>;
+	instanceType: "editor" | "agent";
+	result: "success" | "error" | "stopped";
+};
+
+export type JournalEntry = RawJournalEntry & {
+	title: string;
+	component: Component;
+	componentDefinition: WriterComponentDefinition;
+};
+
+const rawEntries: Ref<Record<string, RawJournalEntry>> = ref({});
+
+const searchText = ref("");
+const selectedStatuses = ref<string[]>([]);
+const selectedTriggers = ref<string[]>([]);
+
+const entries: ComputedRef<Record<string, JournalEntry>> = computed(() => {
+	return Object.fromEntries(
+		Object.entries(rawEntries.value).map(([key, entry]) => {
+			const component = wf.getComponentById(entry.trigger.component.id);
+			const componentDefinition = wf.getComponentDefinition(
+				component.type,
+			);
+
+			const title =
+				component.type === "blueprints_blueprint"
+					? component.content.key
+					: component.content.alias || componentDefinition.name;
+
+			return [key, { ...entry, title, component, componentDefinition }];
+		}),
+	);
+});
+
+const filteredEntries: ComputedRef<Record<string, JournalEntry>> = computed(
+	() => {
+		return Object.fromEntries(
+			Object.entries(entries.value).filter(([_, entry]) => {
+				const searchMatch =
+					searchText.value === "" ||
+					entry.title
+						.toLowerCase()
+						.includes(searchText.value.toLowerCase());
+				const statusMatch =
+					selectedStatuses.value.length === 0 ||
+					selectedStatuses.value.includes(entry.result);
+				const triggerMatch =
+					selectedTriggers.value.length === 0 ||
+					selectedTriggers.value.includes(entry.trigger.type);
+				return searchMatch && statusMatch && triggerMatch;
+			}),
+		);
+	},
+);
+
+const sortedEntries: ComputedRef<Record<string, JournalEntry>> = computed(
+	() => {
+		const sortedArray = Object.entries(filteredEntries.value).sort(
+			([_, a], [__, b]) =>
+				new Date(b.timestamp).getTime() -
+				new Date(a.timestamp).getTime(),
+		);
+
+		return Object.fromEntries(sortedArray);
+	},
+);
+
+async function loadEntries() {
+	const response = await fetch(
+		convertAbsolutePathtoFullURL("/api/data/retrieve"),
+		{
+			method: "POST",
+			headers: {
+				"Content-Type": "application/json",
+			},
+			body: JSON.stringify({
+				key_contains: "wf-journal-",
+				skip_keys: Object.keys(rawEntries.value),
+			}),
+		},
+	);
+
+	if (!response.ok) {
+		pushToast({
+			type: "error",
+			message: "Failed to fetch the execution history",
+		});
+		return;
+	}
+
+	const data = await response.json();
+
+	rawEntries.value = { ...rawEntries.value, ...data.result };
+}
+
+const downloadAsJson = () => {
+	const jsonString = JSON.stringify(filteredEntries.value, null, 2);
+	const blob = new Blob([jsonString], { type: "application/json" });
+
+	const a = document.createElement("a");
+	a.href = URL.createObjectURL(blob);
+	a.download = "agent-journal.json";
+
+	document.body.appendChild(a);
+	a.click();
+	document.body.removeChild(a);
```

Comment:
You need to release the URL in memory `	window.URL.revokeObjectURL(url)`
```suggestion
  const url = URL.createObjectURL(blob);

	const a = document.createElement("a");
	a.href = a.href;
	a.download = "agent-journal.json";

	document.body.appendChild(a);
	a.click();
	document.body.removeChild(a);
	window.URL.revokeObjectURL(url)
```

Reply:
Also a try catch can be good stringify can throw error.

Reply:
```suggestion
	const url = URL.createObjectURL(blob);
	try {
		const a = document.createElement("a");
		a.href = a.href;
		a.download = "agent-journal.json";

		document.body.appendChild(a);
		a.click();
		document.body.removeChild(a);
	} finally {
		window.URL.revokeObjectURL(url)
	}
```

Reply:
I don't expect .stringify to be throwing because we already got all this data from the BE. But I just thought that we probably should not include `component` and `componentDefinition`, just raw entries